designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ProcessActivitySpec.scala

```diff
@@ -0,0 +1,361 @@
+
+          createAttachment(wrongName, "test content")
+            .statusCode(404)
+            .equalsPlainBody(s"No scenario $wrongName found")
+        }
+      }
+
+      "not authenticated should" - {
+        "forbid access" in {
+          given()
+            .multiPartBody(fileContent = "test", fileName = "test.xml")
+            .noAuth()
+            .when()
+            .post(s"$nuDesignerHttpAddress/api/processes/${processName.value}/1/activity/attachments")
+            .Then()
+            .statusCode(401)
+            .equalsPlainBody("The resource requires authentication, which was not supplied with the request")
+        }
+      }
+    }
+
+    "download attachment when" - {
+      "authenticated should" - {
+        "download existing attachment" in {
+          val fileContent = "very important content"
+          createSavedProcess(process, category = Category1, TestProcessingTypes.Streaming)
+          createAttachment(processName, fileContent)
+          val attachmentId = getActivity(processName).extractLong("attachments[0].id")
+
+          getAttachment(processName, attachmentId)
+            .statusCode(200)
+            .equalsPlainBody(fileContent)
+        }
+
+        "return empty body for no existing attachment" in {
+          createSavedProcess(process, category = Category1, TestProcessingTypes.Streaming)
+
+          getAttachment(processName, 1L)
+            .statusCode(200)
+            .equalsPlainBody("")
+        }
+
+        "return 404 for no existing process" in {
+          val wrongName = ProcessName("wrongProcessName")
+
+          getAttachment(wrongName, 1)
+            .statusCode(404)
+            .equalsPlainBody(s"No scenario $wrongName found")
+        }
+      }
+
+      "not authenticated should" - {
+        "forbid access" in {
+          given()
+            .noAuth()
+            .when()
+            .get(s"$nuDesignerHttpAddress/api/processes/${processName.value}/activity/attachments/1")
+            .Then()
+            .statusCode(401)
+            .equalsPlainBody("The resource requires authentication, which was not supplied with the request")
+        }
+      }
+    }
+  }
+
+  private def createComment(processName: ProcessName, commentContent: String): ValidatableResponse = {
+    given()
+      .plainBody(commentContent)
+      .basicAuth("writer", "writer")
+      .when()
+      .post(s"$nuDesignerHttpAddress/api/processes/${processName.value}/1/activity/comments")
+      .Then()
+  }
+
+  private def deleteComment(processName: ProcessName, commentId: Long): ValidatableResponse = {
+    given()
+      .basicAuth("writer", "writer")
+      .when()
+      .delete(s"$nuDesignerHttpAddress/api/processes/${processName.value}/activity/comments/$commentId")
+      .Then()
+  }
+
+  private def getActivity(processName: ProcessName): ValidatableResponse = {
+    given()
+      .contentType(ContentType.JSON.withCharset(StandardCharsets.UTF_8))
```

Comment:
it's a GET request. Do we need to pass the content type header?

Reply:
Done